### ISSUE TITLE
Watch freezes on stop tracking

### ISSUE THREAD

**Issue by MrFly** [urbandroid-team/Sleep-as-Android-Garmin-Addon#19]
I'm using current (today is 10.12.2018) released versions of SaA (beta), Garmin addon, Garmin watch app with a OnePlus 6 on Android Pie with a Garmin VA3.

Every time the tracking is stopped - no difference if it is stopped from the watch or phone - the watch freezes completely. It stucks at the screen "Tracking". I tried to get a log or something, but I were not able to access the watch anymore. The only way to "fix" it is by performing a cold restart (pressing 15 secs on the hardware button).

Edit: Don't know how to reference an issue in the correct way, so: [Mentioned from Artaud](https://github.com/urbandroid-team/Sleep-as-Android-Garmin-Addon/issues/10#issuecomment-445772631)

**Comment by skros89** [comment]
I'm using the same setup as @MrFly. I don't remember if it happened every time for me though, am gonna pay more attention for the next few days.

I don't know if this can serve as a hint, but this morning I stopped tracking from my phone and forgot to reboot the watch immediately. In the "frozen" state, the watch battery drained from ~80 % to 0 in around 15 min.

**Comment by Artaud** [comment]
@skros89 The thing regarding battery is very interesting.

In general I assume this issue is a direct product of the [FAILURE_DURING_TRANSFER Garmin bug](https://forums.garmin.com/forum/developers/connect-iq/connect-iq-bug-reports/158068-failure_during_transfer-issue-again-now-using-comm-sample/page3), so until they fix it, the most we can hope for is some kind of workaround. Does anybody have any ideas? :) 

The problem is probably that the stop tracking command comes in such a long time after the app starts, that it is almost guaranteed that the messages from the watch and to the watch will clash at some point and trigger the bug.

A wild idea for a workaround is making sure that the incoming and outgoing messages are timed in a way they will never clash. This might not be possible however, depending on the environment. And also it may not solve the problem as the exact underlying bug is not precisely known.

**Comment by jgoss1074** [comment]
FYI, this same problem occurs for me on Garmin vivoactive 3 watch...

**Comment by skros89** [comment]
@Artaud This morning stop tracking from the phone correctly stopped the tracking on the watch for me. Only thing different than usual was that I didn't have an alarm enabled. Maybe that can help with a workaround? I'm going to test a few more nights without an alarm see if that's really the difference.

**Comment by jgoss1074** [comment]
@skros89 I noticed yesterday that an update was pushed to my watch by Garmin. I suspect Garmin fixed their bug, though I haven't tested it myself. You may have gotten and update to your watch and didn't notice.

**Comment by Artaud** [comment]
@jgoss1074 Which watch do you have? They update models independently

**Comment by jgoss1074** [comment]
@artaud I have a vivoactive 3 (not music model).

**Comment by skros89** [comment]
@jgoss1074 You're right, I haven't noticed the update apparently. I am on the new VA3 software, guess I'll test it with the alarm tonight to see if it's really fixed.

**Comment by jgoss1074** [comment]
I tried this last night and it worked fine for me when I stopped tracking this morning!

**Comment by skros89** [comment]
Yes, worked perfectly for me the last few days as well. Guess garmin finally fixed their bug on VA3!

**Comment by nmmacmillan** [comment]
Picked up a VA4 and I'm seeing this issue. Stopping tracking on the phone or watch requires a reboot of the watch.

**Comment by futureweb** [comment]
Just had this issue on my Fenix 6x Watch. (Latest FW)
I hit "snooze 10min" on my Phone, then tried to stop/exit the Sleep App on Watch, no chance to get it stopped ... stopping the snooze timer on the Mobile didn't do anything too ... had ro reboot watch to get out of the App.
